### PR TITLE
Feat/save simplified

### PR DESCRIPTION
**Pull request by Jade-ZS**
### What does this PR do? 
I added a save card function. 
- When a user click on a save button, the button should become and stay red.
- When a user click the button again to unsave the item, the background color of the button should become white

### How was it implemented? 
- I used createContext and useContext to store the saved items and functions that help save and unsave.
- I created an isSaved state and checkIfSaved function to add conditional classes to the save button
- I used useEffect to set isSaved to avoid infinite loop

### Checklist:
[ ] Tests have been added or updated to cover changes
[X] Documentation has been updated (if necessary)
[ ] Code follows project coding standards
[ ] Changes have been reviewed by at least one other team member
[ ] Changes have been merged into the main branch

### Related Issues:
I haven't rendered the saved cards on the UI side. For now, the saved items will only be shown in the dev tool console.

### Screenshots: 
Screenshot 2023-08-02 at 10 22 55 AM
Screenshot 2023-08-02 at 10 24 00 AM

### Additional Notes: 
The code is not dry and have many console.log comments for debugging purpose that I haven't gotten rid of. I'll have to do more refactor after I finish implementing the basic features.

### CODE REVIEW

File: src/components/Result/Result.js [hunk 1]
```diff
@@ -8,14 +8,15 @@ export default function Result({result}) {
...
 
-  const checkIfSaved = () => {
-    const output = saves.some(element => JSON.stringify(element) === JSON.stringify(location));
-    console.log('checkIfSaved output: ', output, 'isSaved: ', isSaved)
-    return output;
-  }
-
   useEffect(() => {
+    const checkIfSaved = () => {
+      const output = saves.some(element => JSON.stringify(element) === JSON.stringify(location));
```
Comment (alivaditis) [inline src/components/Result/Result.js:13]: nice use of the some iterator method here

Comment (alivaditis) [inline src/components/Result/Result.js:13]: Cool to see that saves is coming from global context!!!

**Review by alivaditis — APPROVED** [review]
Hey Jade, super cool to see the stuff you're doing with the contextApi.  I had some problems with installing the dependencies, 
`No matching version found for @babel/plugin-proposal-private-property-in-object@^7.22.5.`

But the UI in the screenshot looks super cool and good job getting state to update the new location value.

File: src/components/Result/Result.js (hunk 1 above)
Comment (Jade-ZS) [reply src/components/Result/Result.js:13]: Thank you!

**Comment by Jade-ZS** [comment]
> 

Thank you Alec! I'll look into that. 
Does running this command help? `npm install --save-dev @babel/plugin-proposal-private-property-in-object`

**Pull request merged**